ui/explore/src/components/ExploreManager/TracesExplorer.tsx

```diff
@@ -32,55 +90,21 @@ function TracePanel({ queries }: { queries: QueryDefinition[] }) {
       })
     : [];
 
-  return (
-    <Box height={height}>
-      <DataQueriesProvider definitions={definitions}>
-        <Stack height="100%">
-          <Box height={PANEL_PREVIEW_TRACES_SCATTERPLOT_HEIGHT}>
-            <Panel
-              panelOptions={{
-                hideHeader: true,
-              }}
-              definition={{
-                kind: 'Panel',
-                spec: { queries, display: { name: '' }, plugin: { kind: 'ScatterChart', spec: {} } },
-              }}
-            />
-          </Box>
-          <Panel
-            sx={{ flexGrow: 1, marginTop: '15px' }}
-            panelOptions={{
-              hideHeader: true,
-            }}
-            definition={{
-              kind: 'Panel',
-              spec: { queries, display: { name: '' }, plugin: { kind: 'TraceTable', spec: {} } },
-            }}
-          />
-        </Stack>
-      </DataQueriesProvider>
-    </Box>
-  );
-}
-
-export function TracesExplorer() {
-  const { tab, queries, setTab, setQueries } = useExplorerManagerContext();
+  // TODO: how to access query string here?
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const isSingleTrace = definitions.length === 1 && isValidTraceId((definitions[0]?.spec as any).query || '');
```

Comment:
How can I access the `query` here? It's required to determine whether to show the trace table or Gantt chart.

Reply:
How to access `query` without the `as any` ?

Reply:
I think it's not possible, because the required type `TempoTraceQuerySpec` is in `tempo-plugin`, which is not (and should not be) a dependency of `@perses-dev/explore`. I've extended the condition to check the `kind`s, but I guess that's all I can do here :)

... unless we would move the `query` one level up to the `TraceQuery` instead of `TempoTraceQuery`. But I guess it's ok as-is now.